### PR TITLE
Add GitHub action to check for broken markdown links

### LINKED ISSUE #10439: Broken links in Azure-ARM builders documentation
There are a number of broken links in the Azure-ARM documentation https://www.packer.io/docs/builders/azure-arm

Both appear several times in the page:
https://azure.microsoft.com/en-us/documentation/articles/resource-groups-vm-searching/
https://github.com/hashicorp/packer/tree/master/examples/azure

### PR DESCRIPTION
**Pull request by nywilken**
This changes introduces the [GitHub Action - Markdown link check](https://github.com/gaurav-nelson/github-action-markdown-link-check#github-action---markdown-link-check-%EF%B8%8F) to help in finding potential broken links in the Packer documentation. 

The default configuration is to only check '.mdx' under the directory `website/content/...`. 

On pull-requests only modified files will be checked.
On a regularly scheduled run links will be checked on the mater branch. 

### TODOs
- [x] Test changes against PRs
- [x] Add scheduled job for master branch 
- [x] Address issues raised by markdown link checker
- [x] Validate steps for temporarily updating packer-plugin-sdk partials
- [x] Update README with link checker details

Closes #10439

### CODE REVIEW

**Comment by nywilken** [comment]
cc @jescalan

**Review by azr — APPROVED** [review]
Woa, so many fixed broken links 👍🏼 👍🏼 👍🏼 👍🏼  LGTM.

File: .github/workflows/scheduled-link-checker.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+on:
```
Comment (sylviamoss) [inline .github/workflows/scheduled-link-checker.yml:1]: Is it possible, if not set yet, to sends us a message on slack whenever this fails?

**Review by sylviamoss — APPROVED** [review]
Super nice! 👍🏼

File: website/content/docs/builders/outscale/index.mdx [hunk 2]
```diff
@@ -105,13 +105,13 @@ export OUTSCALE_X509KEY="the/path/to/your/x509key"
 ### Checking that system time is current
 
 Outscale uses the current time as part of the [request signing
-process](http://docs.aws.osc.com/general/latest/gr/sigv4_signing.html). If
+process](https://docs.aws.amazon.com/general/latest/gr/signature-version-4.html). If
```
Comment (SwampDragons) [inline website/content/docs/builders/outscale/index.mdx:108]: does outscale not have their own version of these docs? I know their API directly and intentionally mirrored AWS for a while but it surprises me they didn't also mirror the docs

**Review by SwampDragons — APPROVED** [review]
Amazing job; thanks for taking the time to implement this and track down all those broken links.

File: website/content/docs/builders/outscale/index.mdx (hunk 2 above)
Comment (nywilken) [reply website/content/docs/builders/outscale/index.mdx:108]: Ah good catch. I somehow thought this was pointing outscale. I found this URL https://wiki.outscale.net/display/EN/About+Signatures+of+API+Requests which is probably where this link should point to now.

File: .github/workflows/scheduled-link-checker.yml (hunk 1 above)
Comment (nywilken) [reply .github/workflows/scheduled-link-checker.yml:1]: I'm not sure if this is possible out of the box. But there looks to be some GHA for slack notifications that could help. I say we hold on slack setup for now. 

We will for sure get a GitHub failure email, and I'm hoping that maybe our github Slack integration in #feed-packer-github handles some of the ci failures :crossed_fingers:

**Pull request merged**